cache/cache.go

```diff
@@ -95,14 +97,14 @@ func (nc *lruCache) Len() int {
 }
 
 func (c *lruCache) Remove(key []byte) Node {
-	if elem, exists := c.dict[string(key)]; exists {
+	if elem, exists := c.dict[ibytes.UnsafeBytesToStr(key)]; exists {
```

Comment:
I'm surprised that this is helpful because, from my understanding, copying during `[]byte` to `string` conversion when accessing a map by key should be optimized by the Go compiler.

Sources:
- https://github.com/golang/go/issues/3512
- https://stackoverflow.com/questions/43470284/bytestring-vs-bytestring/43470344#43470344
- https://github.com/golang/go/blob/f5f5a8b6209f84961687d993b93ea0d397f5d5bf/src/cmd/gc/order.c#L943-L953

Reply:
I'm wondering if the benchmark would remain the same as it is right now if the `map[string]` changes are reverted

Reply:
I can check, however I for consistency we can keep the casting here.